tests/python/gaia-ui-tests/gaiatest/apps/search/regions/browser.py

```diff
@@ -38,6 +39,8 @@ def is_page_loading(self):
 
     def wait_for_page_to_load(self, timeout=30):
         Wait(self.marionette, timeout).until(lambda m: not self.is_page_loading)
+        Wait(self.marionette, timeout).until(lambda m: self._root_element.find_element(*self._reload_button_locator)
```

Comment:
You should locate the element outside of the wait, otherwise it will be repeatedly looked up, when all we need to do is check if it's displayed.